### PR TITLE
feat: add `uncaughtException` handler

### PR DESCRIPTION
**Pull request by eduardoboucas**
#### Summary

Adds an `uncaughtException` handler to the process so that we can wrap any uncaught exceptions with some nice messaging explaining that an error with the Netlify CLI itself has occurred. The message also includes useful links and the full system info, ready to be copied and pasted to a bug report.

Screenshot 2023-09-19 at 00 26 11

### CODE REVIEW

**Comment by eduardoboucas** [comment]
@netlify/support-pod-engineering would love a thumbs up from you on the copy.

**Review by estephinson — APPROVED** [review]
:shipit:

File: src/commands/main.mjs [hunk 1]
```diff
@@ -37,16 +37,42 @@ import { createWatchCommand } from './watch/index.mjs'
 
 const SUGGESTION_TIMEOUT = 1e4
 
-const getVersionPage = async () => {
-  // performance optimization - load envinfo on demand
-
-  const data = await envinfo.run({
+process.on('uncaughtException', async (err) => {
+  console.log('')
```
[the remark this answers is not in the document]
Comment (lemusthelroy) [reply src/commands/main.mjs:41]: Actually, I'm guessing this creates a line break in the log and that is why?

[the remark this answers is not in the document]
Comment (eduardoboucas) [reply src/commands/main.mjs:41]: Yes. It's just a way to better separate the error from any messages shown previously.

**Comment by estephinson** [comment]
Future want maybe - some way of adding a error ID of some kind for support cases to be tracked against errors automatically reported to the APM

**Comment by Hrishikesh-K** [comment]
This is so cool!!! Thank you for adding this!

**Pull request merged**